src/RadioButton.tsx

```diff
@@ -0,0 +1,76 @@
+import styled from 'styled-components'
+import React, {InputHTMLAttributes, ReactElement} from 'react'
+import sx, {SxProp} from './sx'
+
+export type RadioButtonProps = {
+  /**
+   * A unique value that is never shown to the user.
+   * Used during form submission and to identify which radio button in a group is selected
+   */
+  value: string
+  /**
+   * Name attribute of the input element. Required for grouping radio inputs
+   */
+  name: string
+  /**
+   * Apply inactive visual appearance to the radio button
+   */
+  disabled?: boolean
+  /**
+   * Indicates whether the radio button is selected
+   */
+  checked?: boolean
+  /**
+   * Forward a ref to the underlying input element
+   */
+  ref?: React.RefObject<HTMLInputElement>
+  /**
+   * Indicates whether the radio button must be checked before the form can be submitted
+   */
+  required?: boolean
+  /**
+   * Indicates whether the radio button validation state is non-standard
+   */
+  validationStatus?: 'error' | 'success' // TODO: hoist to Validation typings
```

Comment:
Don't forget this TODO. Here's where I put them in my PR: https://github.com/primer/react/pull/1611/files#diff-410797b1c4259d74820358a792a62f83cfc7a1c0a0eb26226c89e4621b88d42cR1